tool/internal/instrument/trampoline.go

```diff
@@ -676,37 +687,81 @@ func (ip *InstrumentPhase) rewriteHookContext() {
 	idx := 0
 	if ast.HasReceiver(ip.targetFunc) {
-		recvType := ip.targetFunc.Recv.List[0].Type
+		splitRecv := ast.SplitMultiNameFields(ip.targetFunc.Recv)
+		recvType := replaceTypeParamsWithAny(splitRecv.List[0].Type, ip.targetFunc.Type.TypeParams)
 		clause := setParamClause(idx, recvType)
 		methodSetParamBody.List = append(methodSetParamBody.List, clause)
 		clause = getParamClause(idx, recvType)
 		methodGetParamBody.List = append(methodGetParamBody.List, clause)
 		idx++
 	}
-	for _, param := range ip.targetFunc.Type.Params.List {
-		paramType := desugarType(param)
-		for range param.Names {
-			clause := setParamClause(idx, paramType)
-			methodSetParamBody.List = append(methodSetParamBody.List, clause)
-			clause = getParamClause(idx, paramType)
-			methodGetParamBody.List = append(methodGetParamBody.List, clause)
-			idx++
-		}
+	splitParams := ast.SplitMultiNameFields(ip.targetFunc.Type.Params)
+	for _, param := range splitParams.List {
+		paramType := replaceTypeParamsWithAny(desugarType(param), ip.targetFunc.Type.TypeParams)
+		clause := setParamClause(idx, paramType)
+		methodSetParamBody.List = append(methodSetParamBody.List, clause)
+		clause = getParamClause(idx, paramType)
+		methodGetParamBody.List = append(methodGetParamBody.List, clause)
+		idx++
 	}
 	// Rewrite GetReturnVal and SetReturnVal methods
 	if ip.targetFunc.Type.Results != nil {
 		idx = 0
-		for _, retval := range ip.targetFunc.Type.Results.List {
-			retType := desugarType(retval)
-			for range retval.Names {
-				clause := getReturnValClause(idx, retType)
-				methodGetRetValBody.List = append(methodGetRetValBody.List, clause)
-				clause = setReturnValClause(idx, retType)
-				methodSetRetValBody.List = append(methodSetRetValBody.List, clause)
-				idx++
+		splitResults := ast.SplitMultiNameFields(ip.targetFunc.Type.Results)
+		for _, retval := range splitResults.List {
+			retType := replaceTypeParamsWithAny(desugarType(retval), ip.targetFunc.Type.TypeParams)
+			clause := getReturnValClause(idx, retType)
+			methodGetRetValBody.List = append(methodGetRetValBody.List, clause)
+			clause = setReturnValClause(idx, retType)
+			methodSetRetValBody.List = append(methodSetRetValBody.List, clause)
+			idx++
+		}
+	}
+}
+
+// isTypeParameter checks if a type expression is a bare type parameter identifier
+func isTypeParameter(t dst.Expr, typeParams *dst.FieldList) bool {
+	if typeParams == nil {
+		return false
+	}
+	ident, ok := t.(*dst.Ident)
+	if !ok {
+		return false
+	}
+	// Check if this identifier matches any type parameter name
+	for _, field := range typeParams.List {
+		for _, name := range field.Names {
+			if name.Name == ident.Name {
+				return true
 			}
 		}
 	}
+	return false
+}
+
+// replaceTypeParamsWithAny replaces type parameters with interface{} for use in
+// non-generic contexts like HookContextImpl methods
+func replaceTypeParamsWithAny(t dst.Expr, typeParams *dst.FieldList) dst.Expr {
```

Comment:
replaceTypeParamsWithAny and rectifyAnyType seem to have some similarities. It might be possible to refactor them into a single function, but this can be handled in a follow-up PR.